VideoLocker/src/main/java/org/edx/mobile/view/EditUserProfileFragment.java

```diff
@@ -161,18 +170,9 @@ public boolean onMenuItemClick(MenuItem item) {
                                 break;
                             }
                             case R.id.remove_photo: {
-                                executePhotoTask(new DeleteAccountImageTask(getActivity(), username) {
-                                    @Override
-                                    protected void onSuccess(Void aVoid) throws Exception {
-                                        hideLoading();
-                                    }
-
-                                    private void hideLoading() {
-                                        if (null != viewHolder) {
-                                            viewHolder.profileImageProgress.setVisibility(View.GONE);
-                                        }
-                                    }
-                                });
+                                final Task task = new DeleteAccountImageTask(getActivity(), username);
```

Comment:
Since `SetAccountImageTask` [can't][1] be converted into a `Call`, `DeleteAccountImageTask` needs to remain an instance of `Task` too, in order for them both to be easily assigned to a single variable with the same cancelling semantics (to be able to easily cancel and replace each other). Otherwise, there would need to be some special logic written to handle both of these types/variables. It would probably be better to switch to RxJava instead, which could probably be used with the Retrofit/OkHttp asynchronous API to provide equivalent dynamics.

[1]: https://github.com/edx/edx-app-android/pull/799#discussion_r77359787